universal-login-react/src/ui/commons/ScrollProgressDots.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+interface ScrollProgressDotsProps {
+  dotOpacities: Array<number>;
+  minimumOpacity?: number;
+  maximumOpacity?: number;
+}
+
+export const ScrollProgressDots = ({dotOpacities, minimumOpacity = 0.0, maximumOpacity = 1.0}: ScrollProgressDotsProps) => {
+  dotOpacities = dotOpacities.map((opacity) => opacity < minimumOpacity ? minimumOpacity : opacity);
+  dotOpacities = dotOpacities.map((opacity) => opacity > maximumOpacity ? maximumOpacity : opacity);
```

Comment:
Maybe one map, something like this (maybe less sophisticated but you know what is going on at glance)
```js
dotOpacities = dotOpacities.map((opacity) => {
    if (opacity < minimumOpacity) {
          return minimumOpacity
     } else if (opacity > maximumOpacity) {
          return maximumOpacity
     }
     return opacity
}
```

Reply:
```
const clamp = (val, min, max) => Math.max(Math.min(val, max), min)

 const dots = dotOpacities.map((opacity, i) =>
    <div key={i} style={{opacity: clamp(opacity, min, max)}} className='scroll-progress-dot'/>,
  );

Reply:
And place `clamp` in some utils

Reply:
In a follow-up PR